src/main/java/com/amazon/opendistroforelasticsearch/security/OpenDistroSecurityPlugin.java

```diff
@@ -1010,10 +1015,10 @@ public void onNodeStarted() {
         }
         final Set<ModuleInfo> securityModules = ReflectionHelper.getModulesLoaded();
         log.info("{} Open Distro Security modules loaded so far: {}", securityModules.size(), securityModules);
-        if(complianceConfig != null && complianceConfig.isEnabled() && complianceConfig.isLogExternalConfig() && !complianceConfig.isExternalConfigLogged()) {
-        	log.info("logging external config");
-        	auditLog.logExternalConfig(complianceConfig.getSettings(), complianceConfig.getEnvironment());
-            complianceConfig.setExternalConfigLogged(true);
+        if(complianceConfig != null && complianceConfig.isEnabled() && complianceConfig.shouldLogExternalConfig() && !externalConfigLogged) {
+            log.info("logging external config");
+            auditLog.logExternalConfig(settings, environment);
+            externalConfigLogged = true;
```

Comment:
Possible race condition, two threads may race for `externalConfigLogged`.

Reply:
externalConfigLogged is set t volatile. This is existing behavior as well. It is just pulled out of compliance config

Reply:
`volatile` does not prevent race condition, it only provides memory barrier. It will be OK to file that as an issue and address in a separate PR.

Reply:
Yes it would be great to address it separately.
Are you saying onNodeStarted will called by multiple threads ? Need to investigate the race condition

Reply:
It is a *possible* race condition. As plugin has no control when and how `onNodeStarted()` is called, it will be safer to avoid the race condition, especially that it will not cost anything. Just replace `volatile` with `AtomicBoolean` and use `getAndSet()`.